lgn-worker/src/main.rs

```diff
@@ -296,23 +275,38 @@ async fn run_worker(
 
     loop {
         debug!("Waiting for message...");
-        tokio::select! {
-            Some(inbound_message) = inbound.next() => {
-                let msg = match inbound_message {
-                    Ok(msg) => msg,
-                    Err(e) => {
-                        bail!("connection to the gateway ended with status: {e}");
-                    }
-                };
-                let result = process_message_from_gateway(&mut provers_manager, msg, &mut outbound, &mp2_requirement).await;
-                if result.is_ok() {
-                    last_task_processed.store(SystemTime::now().duration_since(UNIX_EPOCH)?.as_secs(), Ordering::Relaxed);
-                }
-                if let Err(e) = result {
-                    bail!("task processing failed: {e:?}");
```

Comment:
This is the main change, now instead of bailing a metric is increased and the next message is processed.